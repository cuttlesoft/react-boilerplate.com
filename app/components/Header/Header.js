/* eslint-disable react/prop-types */
import React from 'react'

// Components
import { Header as GrommetHeader } from 'grommet'

// Helpers
import { doc } from './Header.doc'
import { createWithDoc } from '../../utils/helpers'

/**
 *
 * Header
 *
 */
const Header = ({ children, ...rest }) => <GrommetHeader {...rest}>{children}</GrommetHeader>

export default createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: Header,
})
