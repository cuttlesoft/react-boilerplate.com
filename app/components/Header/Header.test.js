import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import Header from './Header'

const renderComponent = (props = {}) => render(<Header {...props}>TESTING</Header>)

/**
 *
 * Tests for Header
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */
describe('Header', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
