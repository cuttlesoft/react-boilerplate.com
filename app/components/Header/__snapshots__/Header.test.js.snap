// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 fXgXKw"
  >
    <header
      class="StyledBox-sc-13pk1d4-0 gmyXNm"
      style="width: 100vw;"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bFwQzJ"
      >
        <a
          class="Link__StyledLink-sc-1vpl4pu-0 gEErvb"
          href="/"
          theme="[object Object]"
        >
          React Boilerplate
        </a>
      </div>
    </header>
  </div>
</div>
`;
